core/src/test/java/org/apache/calcite/sql/test/SqlOperatorBaseTest.java

```diff
@@ -7482,11 +7482,6 @@ private void checkNullOperand(SqlTester tester, String op) {
         "Array['foo', 'bar']",
         "[foo, bar]",
         "CHAR(3) NOT NULL ARRAY NOT NULL");
-
-    // empty array is illegal per SQL spec. presumably because one can't
```

Comment:
why are we allowing a violation to SQL specs?

Reply:
I think it is not a violation in hive SQL, we can run 
```
select array() as a;
```
in hive.
And in some production views, users used it to create a view.

Reply:
I see. Are production views using `Array[]` in where clauses (e.g., `where column = Array[]`)? I don't see where else would it make sense.
If indeed they are used in filters, don't we need to type the elements so that the comparison is carried out properly? Can you verify that running a query with `where column = Array[]` works as expected for rows with and without `Array[]`?

Reply:
No, production views are using `Array[]` as a select item to union with another array column:
```
SELECT ARRAY() AS x FROM table1
UNION ALL
SELECT array_column_of_table2 AS x FROM table2
```
where `table1` doesn't have such an `array_column`.

In the production view, users don't care about the type of the elements, they just want to do the union.

Reply:
I see.
Can you test your changes for `Array[]` in the where clause?
Can you also add unit tests for both cases?

Reply:
Array can not be used with `=` in where clause, if we just run:
```
hive (default)> select 1 from table where arr=array();
FAILED: SemanticException [Error 10016]: Line 1:40 Argument type mismatch 'arr': The 1st argument of EQUAL  is expected to a primitive type, but list is found
```
because both arguments of `=` should be primitive type rather than list. And all the `ARRAY()` in production views are used as select item.
As for the unit test, I think it is better to add them to coral while bumping calcite version so that we can see the SQL with `ARRAY()` could be translated successfully with this fix (in calcite, we don't have infra to do the translation)

Reply:
Thanks @ljfgem, the plan sounds good to me. Do you have further concerns regarding this PR @shenodaguirguis ?